pkg/sql/parser/normalize.go

```diff
@@ -718,19 +724,25 @@ func (v *isConstVisitor) run(expr Expr) bool {
 	return v.isConst
 }
 
-func isVar(expr Expr) bool {
-	_, ok := expr.(VariableExpr)
-	return ok
+func isVar(evalCtx *EvalContext, expr Expr) bool {
+	switch expr.(type) {
+	case VariableExpr:
+		return true
+	case *Placeholder:
+		return evalCtx != nil && (!evalCtx.HasPlaceholders() || evalCtx.Placeholders.IsUnresolvedPlaceholder(expr))
```

Comment:
If I understand this change, maybe it would make sense to rename this function something like `isNonConstant`? I know "variable" as an adjective implies that but it's easy to misread as the noun in this case (or even just a comment implying that "variable" in this context refers to the expression varying, rather than a `VariableExpr`).